src/DssDirectDepositHub.sol

```diff
@@ -0,0 +1,459 @@
+        pool.withdraw(total);
+        daiJoin.join(address(this), total);
+
+        // normalized debt == erc20 DAI to pool (Vat rate for this ilk fixed to 1 RAY)
+
+        if (mode == Mode.NORMAL) {
+            vat.frob(ilk, address(pool), address(pool), address(this), -int256(amount), -int256(amount));
+            vat.slip(ilk, address(pool), -int256(amount));
+            vat.move(address(this), vow, _mul(fees, RAY));
+        } else if (mode == Mode.MODULE_CULLED) {
+            vat.slip(ilk, address(pool), -int256(amount));
+            vat.move(address(this), vow, _mul(total, RAY));
+        } else {
+            // This can be done with the assumption that the price of 1 collateral unit equals 1 DAI.
+            // That way we know that the prev End.skim call kept its gap[ilk] emptied as the CDP was always collateralized.
+            // Otherwise we couldn't just simply take away the collateral from the End module as the next line will be doing.
+            vat.slip(ilk, address(end_), -int256(amount));
+            vat.move(address(this), vow, _mul(total, RAY));
+        }
+
+        emit Unwind(ilk, amount, fees);
+    }
+
+    function exec(bytes32 ilk_) external {
+        D3MPoolLike pool = ilks[ilk_].pool;
+
+        pool.accrueIfNeeded();
+        uint256 availableAssets = pool.maxWithdraw();
+        uint256 currentAssets = pool.assetBalance();
+
+        if (vat.live() == 0) {
+            // MCD caged
+            require(end.debt() == 0, "DssDirectDepositHub/end-debt-already-set");
+            require(ilks[ilk_].culled == 0, "DssDirectDepositHub/module-has-to-be-unculled-first");
+            _unwind(
+                ilk_,
+                pool,
+                type(uint256).max,
+                availableAssets,
+                Mode.MCD_CAGED,
+                currentAssets
+            );
+        } else if (ilks[ilk_].tic != 0) {
+            // pool caged
+            _unwind(
+                ilk_,
+                pool,
+                type(uint256).max,
+                availableAssets,
+                ilks[ilk_].culled == 1
+                ? Mode.MODULE_CULLED
+                : Mode.NORMAL,
+                currentAssets
+            );
+        } else {
+            // Determine if it needs to unwind due to debt ceilings
+            (uint256 Art,,, uint256 line,) = vat.ilks(ilk_);
+            uint256 lineWad = line / RAY; // Round down to always be under the actual limit
+            uint256 Line = vat.Line();
+            uint256 debt = vat.debt();
+            uint256 toUnwind;
+            if (Art > lineWad) {
+                toUnwind = Art - lineWad;
+            }
+            if (debt > Line) {
+                toUnwind = _max(toUnwind, _divup(debt - Line, RAY));
+            }
+
+            // Determine if it needs to unwind due plan
+            uint256 targetAssets = ilks[ilk_].plan.getTargetAssets(currentAssets);
+            // Determine if the pool limits our total assets
```

Comment:
This comment's meaning and location is not clear to me, I think we can remove it.

Reply:
it got left behind when i was moving the `maxDeposit` stuff. I fixed it